cocotb_build_libs.py

```diff
@@ -245,7 +245,7 @@ def _get_common_lib_ext(include_dir, share_lib_dir):
     libsim = Extension(
         os.path.join("cocotb", "simulator"),
         include_dirs=[include_dir],
-        libraries=["cocotbutils", "gpilog", "gpi"],
```

Comment:
This makes me a little uneasy - if I'm importing `cocotb.simulator` from a python terminal, I'd rather not pull in the shared library that tries to embed a python interpreter.

Reply:
This isn't an issue when importing cocotb from a Python interpreter, libpython is already loaded at that point (or if it's static, is already defined).

This is due to the fact that there is no shared library between the simulator module and the embedding library, the Python API to update the log level has to be in the extension module, and the Python logger implementation is located in the embed library.

If this is an issue, I can factor the Python logger out into another library. I'm also wondering if we could merge the extension module and the embedding library into one library to avoid the need for a 3rd utility library.

Reply:
@eric-wieser https://drive.google.com/file/d/1fFskjmbM695NIwschUPsGzr554n6AtZ1/view?usp=sharing

Reply:
Went in another direction.